controllers/api/file/PKPManageFileApiHandler.php

```diff
@@ -88,6 +90,51 @@ public function deleteFile($args, $request)
         return \PKP\db\DAO::getDataChangedEvent();
     }
 
+	/**
+	 * Restore original file when cancelling the upload wizard
+	 */
+	public function cancelFileUpload(array $args, Request $request): JSONMessage
+	{
+		if (!$request->checkCSRF()) {
+			return new JSONMessage(false);
+		}
+
+		$submissionFile = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION_FILE);
+		$originalFile = $request->getUserVar('originalFile') ? (array) $request->getUserVar('originalFile') : null;
+		$revisedFileId = $request->getUserVar('fileId') ? (int) $request->getUserVar('fileId') : null;
+
+		// Get revisions and check file IDs
+		$revisions = Repo::submissionFile()->getRevisions($submissionFile->getId());
+		$revisionIds = [];
+		foreach ($revisions as $revision) {
+			$revisionIds[] = $revision->fileId;
+		}
+
+		if (!$revisedFileId && !in_array($revisedFileId, $revisionIds)) {
+			return new JSONMessage(false);
+		}
+
+		if (!isset($originalFile['fileId']) && !in_array($originalFile['fileId'], $revisionIds)) {
+			return new JSONMessage(false);
+		}
+
+		// Restore original submission file
+		Repo::submissionFile()->edit(
+			$submissionFile,
+			[
+				'fileId' => $originalFile['fileId'],
+				'name' => $originalFile['name'],
+				'uploaderUserId' => $originalFile['uploaderUserId'],
```

Comment:
Again, I'm wary of using request data here, because there's nothing enforcing it to be the original user id. It might be that we need to store this information along with revisions so we can retrieve it safely.

You may also be able to grab this data from the submission event log, which will have an entry for the last revision of this submission file.

Reply:
Thanks, done!